Benchmarks/ReflectionPerf.cs

```diff
@@ -0,0 +1,35 @@
+﻿using BenchmarkComponent;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmarks
+{
+    [MemoryDiagnoser]
+    public class ReflectionPerf
+    {
+        ClassWithMarshalingRoutines instance;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            instance = new ClassWithMarshalingRoutines();
+        }
+
+        [Benchmark]
+        public void ExecuteComWrappersCreateKeyValuePairsFactory()
+        {
+            var value = instance.DefaultKeyValuePairProperty;
```

Comment:
We should have all these functions return the value to prevent the variable from being optimized.